probes/unsafeblock/impl_test.go

```diff
@@ -0,0 +1,514 @@
+						Effort: 2,
+					},
+					Location: &finding.Location{Path: "testdata/unsafe.go", LineStart: toUintPointer(4)},
+				},
+			},
+			err: nil,
+		},
+		{
+			name: "All Languages - unsafe",
+			repoLanguages: []clients.Language{
+				{Name: clients.All, NumLines: 0},
+			},
+			filenames: []string{
+				"testdata/unsafe.go",
+				"testdata/unsafe.csproj",
+			},
+			expected: []finding.Finding{
+				{
+					Probe:   Probe,
+					Message: "Golang code uses the unsafe package",
+					Outcome: finding.OutcomeFalse,
+					Remediation: &finding.Remediation{
+						Text:   "Visit the OpenSSF Memory Safety SIG guidance on how to make your project memory safe.\nGuidance for [Memory-Safe By Default Languages](https://github.com/ossf/Memory-Safety/blob/main/docs/best-practice-memory-safe-by-default-languages.md)\nGuidance for [Non Memory-Safe By Default Languages](https://github.com/ossf/Memory-Safety/blob/main/docs/best-practice-non-memory-safe-by-default-languages.md)",
+						Effort: 2,
+					},
+					Location: &finding.Location{Path: "testdata/unsafe.go", LineStart: toUintPointer(4)},
+				},
+				{
+					Probe:   Probe,
+					Message: "C# project file allows the use of unsafe blocks",
+					Outcome: finding.OutcomeFalse,
+					Remediation: &finding.Remediation{
+						Text:   "Visit the OpenSSF Memory Safety SIG guidance on how to make your project memory safe.\nGuidance for [Memory-Safe By Default Languages](https://github.com/ossf/Memory-Safety/blob/main/docs/best-practice-memory-safe-by-default-languages.md)\nGuidance for [Non Memory-Safe By Default Languages](https://github.com/ossf/Memory-Safety/blob/main/docs/best-practice-non-memory-safe-by-default-languages.md)",
+						Effort: 2,
+					},
+					Location: &finding.Location{Path: "testdata/unsafe.csproj"},
+				},
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			raw := &checker.CheckRequest{}
+			mockRepoClient := mockrepo.NewMockRepoClient(ctrl)
+			mockRepoClient.EXPECT().ListProgrammingLanguages().DoAndReturn(func() ([]clients.Language, error) {
+				return tt.repoLanguages, nil
+			}).AnyTimes()
+			mockRepoClient.EXPECT().ListFiles(gomock.Any()).DoAndReturn(func(predicate func(string) (bool, error)) ([]string, error) {
+				return tt.filenames, nil
+			}).AnyTimes()
+			mockRepoClient.EXPECT().GetFileReader(gomock.Any()).DoAndReturn(func(file string) (io.ReadCloser, error) {
+				return os.Open(file)
+			}).AnyTimes()
+			raw.RepoClient = mockRepoClient
+			raw.Dlogger = checker.NewLogger()
```

Comment:
normally in tests we use the TestDetailLogger
https://github.com/ossf/scorecard/blob/3b42b6e7d8c66543d10c7f5ac5d3ecc0d1e56e5a/utests/utlib.go#L38-L41